pkg/register/document.go

```diff
@@ -69,6 +69,30 @@ type RegisterDocument struct {
 	Metadata               Metadata                  `json:"metadata,omitempty"`
 }
 
+func (key RegisterPublicKey) ExistsInSlice(cmp []RegisterPublicKey) error {
+	for _, newKey := range cmp {
+		if newKey.ID == key.ID && newKey.PublicKeyBase58 == key.PublicKeyBase58 && newKey.Type == key.Type {
+			if key.Revoked && !newKey.Revoked {
+				return errors.New("cannot un-revoke key " + key.ID)
+			}
+			return nil
+		}
+	}
+	return errors.New("key " + key.ID + " cannot be removed, only revoked")
+}
+
+func (key RegisterDelegationProof) ExistsInSlice(cmp []RegisterDelegationProof) error {
```

Comment:
should this be `func (proof RegisterDelegationProof)` & error messages updated to say proof instead of key?

Reply:
689aee0f763917eb0ca06eda79d716359a98ac44